src/compiler/checker.ts

```diff
@@ -2658,23 +2699,61 @@ module ts {
             }
         }
 
+        function createNumberIndex(value: Type, index?: Type): IndexType {

```

Comment:
Updated patch is more consistent, renamed 'IndexType' to 'ObjectIndex':
https://github.com/Microsoft/TypeScript/compare/master...jbondc:issue-2491#diff-4b8bd1eea29904f1be39cd864e1a45c0R1532